File: 2486-most-frequent-even-element/most-frequent-even-element.java
class Solution {
    public int mostFrequentEven(int[] nums) {
        HashMap<Integer,Integer> freq=new HashMap<>();
        for(int num: nums){
            freq.put(num,freq.getOrDefault(num,0)+1);
        }
        int maxFreqSoFar=0,finalAns=-1;
        for(Integer num:freq.keySet()){
            int curFreq=freq.get(num);
            if(num%2==1 || curFreq< maxFreqSoFar){
                continue;
            }
            if(curFreq>maxFreqSoFar||num<finalAns){
                maxFreqSoFar=curFreq;
                finalAns=num;
            }
        }
        return finalAns;
    }
}

File: 2486-most-frequent-even-element/most-frequent-even-element.cpp
class Solution {
public:
    int mostFrequentEven(vector<int>& nums) {
        unordered_map<int,int> m;
        int n=nums.size();
        for(int i=0 ; i<n ; i++){
            if(nums[i]%2==0){
                m[nums[i]]++;
            }
        }
        int mostFrequent=-1;
        int highestFrequency=0;
        for(auto it=m.begin(); it!=m.end() ; it++){
            if(highestFrequency <= it->second){
                if(highestFrequency == it->second){
                    if(mostFrequent > it->first){
                        mostFrequent=it->first;
                    }
                }
                else{
                    mostFrequent=it->first;
                    highestFrequency=it->second;
                }
            }
        }
        
        return mostFrequent;
    }
};